miniscope_io/uart_daq.py

```diff
@@ -1,211 +1,351 @@
-import os
 import argparse
-import serial
+import logging
 import multiprocessing
-import coloredlogs, logging
-from datetime import datetime
-import numpy as np
-import cv2
+import os
 import sys
 import time
+from datetime import datetime
+
+import coloredlogs
+import cv2
+import numpy as np
+import serial
+from bitstring import Array, BitArray, Bits
+from routine.ok_utils import okDev
 
 # Parsers for daq inputs
 daqParser = argparse.ArgumentParser("uart_image_capture")
-daqParser.add_argument('port', help="serial port")
-daqParser.add_argument('baudrate', help="baudrate")
+daqParser.add_argument("port", help="serial port")
+daqParser.add_argument("baudrate", help="baudrate")
 
 # Parsers for update LED
 updateDeviceParser = argparse.ArgumentParser("updateDevice")
-updateDeviceParser.add_argument('port', help="serial port")
-updateDeviceParser.add_argument('baudrate', help="baudrate")
-updateDeviceParser.add_argument('module', help="module to update")
-updateDeviceParser.add_argument('value', help="LED value")
+updateDeviceParser.add_argument("port", help="serial port")
+updateDeviceParser.add_argument("baudrate", help="baudrate")
+updateDeviceParser.add_argument("module", help="module to update")
+updateDeviceParser.add_argument("value", help="LED value")
+
 
 class uart_daq:
-    def __init__(self, frame_width: int = 304, frame_height: int = 304, preamble = b'\x78\x56\x34\x12'):
+    def __init__(
+        self,
+        frame_width: int = 304,
+        frame_height: int = 304,
+        preamble=b"\x12\x34\x56",
+        header_fmt={
```

Comment:
will add in docs later but these tuples are basically start and end bit position of a number in header. Not sure if that fits the definition of Range? (at least it should be start/end rather than min/max?)